### ISSUE TITLE
Is it possible to disable locking?

### ISSUE THREAD

**Issue by MarcusWichelmann** [FubarDevelopment/WebDavServer#33]
I'm developing a custom FileSystem implementation for an own virtual filesystem and don't want to support any locking.
Is it possible to completely disable the locking support?

For now I've written following dummy-implementation. But I'm unsure if this will be stable and if there is a much better (and cleaner) way to do that.

Dummy-Implementation:
``` C#
public class DummyLockManager : ILockManager
{
    public int Cost => 0;

    public event EventHandler<LockEventArgs> LockAdded;
    public event EventHandler<LockEventArgs> LockReleased;

    public Task<IEnumerable<IActiveLock>> GetAffectedLocksAsync(string path, bool findChildren, bool findParents, CancellationToken cancellationToken)
        => Task.FromResult(Enumerable.Empty<IActiveLock>());

    public Task<IEnumerable<IActiveLock>> GetLocksAsync(CancellationToken cancellationToken)
        => Task.FromResult(Enumerable.Empty<IActiveLock>());

    public Task<LockResult> LockAsync(ILock l, CancellationToken cancellationToken)
        => Task.FromResult<LockResult>(null);

    public Task<IImplicitLock> LockImplicitAsync(IFileSystem rootFileSystem, IReadOnlyCollection<IfHeaderList> ifHeaderLists, ILock lockRequirements, CancellationToken cancellationToken)
        => Task.FromResult<IImplicitLock>(null);

    public Task<LockRefreshResult> RefreshLockAsync(IFileSystem rootFileSystem, IfHeader ifHeader, TimeSpan timeout, CancellationToken cancellationToken)
        => Task.FromResult<LockRefreshResult>(null);

    public Task<LockReleaseStatus> ReleaseAsync(string path, Uri stateToken, CancellationToken cancellationToken)
        => Task.FromResult(LockReleaseStatus.Success);
}
```

**Comment by fubar-coder** [comment]
TBH: I never thought about disabling locking. If this works for you, then I guess it's fine to go on with it. The main problem is, that we have both implicit and explicit locks. You may provide unit tests to ensure that the implicit locks work as expected with your dummy implementation, but the explicit locks heavily depend on the WebDAV clients behavior.

Why don't you use the in-memory implementation of the lock manager? Those locks aren't really file system locks, but are instead completely managed by the WebDAV server.

**Comment by MarcusWichelmann** [comment]
Okay, thank you!

My filesystem is not only accessible by WebDAV but also by SMB (Samba) and other protocols. So these locks don't affect any of the other protocols anyway. And the fact that it isn't really possible to synchronize them between e.g. Samba and this WebDav library made me thinking about disabling them completely to not have them in an inconsistent state.

But you're right, instead of disabling them, I could keep them at least WebDAV-internal using some in-memory implementation.
I think I'll need to develop my own Redis-based implementation for ILockManager then, because I've multiple processes of the application running as cluster and need to synchronize the state like I've to with the PropertyManager, too.

**Comment by fubar-coder** [comment]
Fixed in the release/2.0 branch
